test/opengl/sum_reduction.cpp

```diff
@@ -33,22 +31,15 @@ int main() {
     result.copy_to_host();
 
     // Check the result.
-    for (int c = 0; c < result.channels(); c++) {
-        for (int y = 0; y < result.height(); y++) {
-            for (int x = 0; x < result.width(); x++) {
-                float temp = 0.0f;
-                for (int r = 0; r < 5; r++){
-                    temp += input(std::min(x+r, input.width()-1), y, c);
-                }
-                float correct = temp / 10.0f * 255.0f;
-                if (fabs(result(x, y, c) - correct) > 1e-3) {
-                    fprintf(stderr, "result(%d, %d, %d) = %f instead of %f\n",
-                            x, y, c, result(x, y, c), correct);
-                    return 1;
-                }
+    if (!Testing::check_result<float>(result, [&](int x, int y, int c) {
+            float temp = 0.0f;
+            for (int r = 0; r < 5; r++) {
+                temp += input(std::min(x + r, input.width() - 1), y, c);
             }
-        }
-    }
+            return temp / 10.0f * 255.0f;
+        },
+                                      1e-3))
```

Comment:
In some way, once past the initial shock of this, it's actually not that unreasonable.  Having (optional) parameters after a lambda block is an awkward and hard-to-parse thing in any case, and  this certainly makes it clear what's going on.

...and having written that, it becomes clear that I should get rid of the awkward code.  

So I moved the tolerance parameter before the block.  And made it optional via an overloaded two-parameter template that calls the three-parameter version with 0 as the second argument.

That eases the eyeball pain :)